Leverage the `RefPtr<T>(T&)` constructor in more places

cdumez opened the pull request:
#### 3c7fe87417430cf1feec93079e4c4576eee0264a
<pre>
Leverage the `RefPtr&lt;T&gt;(T&amp;)` constructor in more places
<a href="https://bugs.webkit.org/show_bug.cgi?id=294309">https://bugs.webkit.org/show_bug.cgi?id=294309</a>

Reviewed by Anne van Kesteren.

* Source/WebCore/Modules/WebGPU/Implementation/WebGPUCompositorIntegrationImpl.h:
* Source/WebCore/Modules/applepay/PaymentCoordinator.cpp:
(WebCore::PaymentCoordinator::beginPaymentSession):
* Source/WebCore/Modules/highlight/AppHighlightStorage.cpp:
(WebCore::findNodeStartingAtPathComponentIndex):
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::startVersionChangeTransaction):
* Source/WebCore/Modules/indexeddb/IDBRequest.cpp:
(WebCore::IDBRequest::setVersionChangeTransaction):
(WebCore::IDBRequest::dispatchEvent):
(WebCore::IDBRequest::willIterateCursor):
* Source/WebCore/Modules/indexeddb/IDBTransaction.cpp:
(WebCore::IDBTransaction::abortDueToFailedRequest):
(WebCore::IDBTransaction::completeNoncursorRequest):
(WebCore::IDBTransaction::completeCursorRequest):
* Source/WebCore/Modules/indexeddb/server/MemoryIndex.cpp:
(WebCore::IDBServer::MemoryIndex::writeTransactionStarted):
* Source/WebCore/Modules/indexeddb/server/MemoryObjectStore.cpp:
(WebCore::IDBServer::MemoryObjectStore::writeTransactionStarted):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::savePreviouslySelectedTextTrackIfNecessary):
* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::MediaSession::MediaSession):
* Source/WebCore/Modules/mediasession/MediaSessionCoordinatorPrivate.cpp:
(WebCore::MediaSessionCoordinatorPrivate::setLogger):
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
* Source/WebCore/Modules/mediastream/RTCEncodedFrame.cpp:
(WebCore::RTCEncodedFrame::setData):
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.cpp:
(WebCore::RTCRtpScriptTransform::initializeTransformer):
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::SpeechRecognition):
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/Modules/webaudio/AudioWorkletNode.cpp:
(WebCore::AudioWorkletNode::setProcessor):
* Source/WebCore/Modules/webaudio/AudioWorkletNode.h:
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::setPeriodicWave):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
(WebCore::WebCodecsVideoFrame::clone):
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::partialOrder):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::handleChildrenChanged):
* Source/WebCore/accessibility/AXTextMarker.cpp:
(WebCore::Accessibility::findObjectWithRuns):
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityRowRange]):
* Source/WebCore/accessibility/playstation/AXObjectCachePlayStation.cpp:
(WebCore::AXObjectCache::postPlatformNotification):
* Source/WebCore/animation/KeyframeEffect.cpp:
(WebCore::KeyframeEffect::setTarget):
* Source/WebCore/animation/StyleOriginatedTimelinesController.cpp:
(WebCore::StyleOriginatedTimelinesController::determineTreeOrder):
(WebCore::StyleOriginatedTimelinesController::updateTimelineForTimelineScope):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::collectIsolatedContexts):
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
(WebCore::ScriptModuleLoader::fetch):
* Source/WebCore/bindings/js/WorkerScriptFetcher.h:
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcLocalValue::setSVGFontFaceElement):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serializeBorderRadius const):
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.cpp:
(WebCore::CSSCalc::evaluateContainerProgress):
* Source/WebCore/dom/Attr.cpp:
(WebCore::Attr::attachToElement):
* Source/WebCore/dom/ChildListMutationScope.cpp:
(WebCore::ChildListMutationAccumulator::childAdded):
* Source/WebCore/dom/ContainerNode.cpp:
(WebCore::containsIncludingHostElements):
(WebCore::dispatchChildInsertionEvents):
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
(WebCore::notifyNodeInsertedIntoDocument):
(WebCore::notifyNodeInsertedIntoTree):
(WebCore::notifyNodeRemovedFromDocument):
(WebCore::notifyNodeRemovedFromTree):
* Source/WebCore/dom/ContentVisibilityDocumentState.cpp:
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyForScrollIfNeeded):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::invalidateQuerySelectorAllResults):
(WebCore::Document::invalidateQuerySelectorAllResultsForClassAttributeChange):
(WebCore::Document::processBaseElement):
(WebCore::Document::idbConnectionProxy):
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::willEnterFullscreen):
(WebCore::documentsToUnfullscreen):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::topmostPopoverAncestor):
* Source/WebCore/dom/Event.cpp:
(WebCore::Event::setEventPath):
(WebCore::Event::timeStampForBindings const):
* Source/WebCore/dom/EventPath.cpp:
(WebCore::RelatedNodeRetargeter::RelatedNodeRetargeter):
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::rangeOfStringInRange):
* Source/WebCore/dom/FragmentDirectiveUtilities.cpp:
(WebCore::FragmentDirectiveUtilities::nearestBlockAncestor):
* Source/WebCore/dom/InternalObserverDrop.cpp:
* Source/WebCore/dom/InternalObserverFilter.cpp:
* Source/WebCore/dom/InternalObserverMap.cpp:
* Source/WebCore/dom/InternalObserverTake.cpp:
* Source/WebCore/dom/LoadableModuleScript.cpp:
(WebCore::LoadableModuleScript::notifyLoadCompleted):
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::surroundContents):
* Source/WebCore/dom/RejectedPromiseTracker.cpp:
(WebCore::RejectedPromiseTracker::reportUnhandledRejections):
* Source/WebCore/dom/UserGestureIndicator.cpp:
(WebCore::UserGestureToken::UserGestureToken):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToNodeRange):
(WebCore::ApplyStyleCommand::surroundNodeRangeWithElement):
(WebCore::ApplyStyleCommand::applyInlineStyleChange):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::wrappingStyleForSerialization):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::appliedEditing):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::DragCaretController::setCaretPosition):
(WebCore::FrameSelection::associateLiveRange):
* Source/WebCore/editing/MarkupAccumulator.cpp:
(WebCore::MarkupAccumulator::serializeNodesWithNamespaces):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::nextInPreOrderCrossingShadowBoundaries):
(WebCore::shouldEmitNewlineAfterNode):
(WebCore::TextIterator::emitText):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::isInUserAgentShadowRootOrHasEditableShadowAncestor):
* Source/WebCore/editing/markup.cpp:
(WebCore::UserSelectNoneStateCache::computeState):
(WebCore::StyledMarkupAccumulator::traverseNodesForSerialization):
(WebCore::highestAncestorToWrapMarkup):
* Source/WebCore/fileapi/FileReader.cpp:
(WebCore::FileReader::readInternal):
* Source/WebCore/html/HTMLFrameOwnerElement.cpp:
(WebCore::SubframeLoadingDisabler::canLoadFrame):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::checkedRadioButtonForGroup const):
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::mapMouseEvent):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::speakCueText):
(WebCore::HTMLMediaElement::sourceWasAdded):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::recalcListItems const):
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::setContainerAndOffsetForRange):
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::computeIsDisabledByFieldsetAncestor const):
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::beginQuery):
* Source/WebCore/html/canvas/WebGLTransformFeedback.cpp:
(WebCore::WebGLTransformFeedback::setProgram):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::insert):
(WebCore::HTMLConstructionSite::insertTextNode):
(WebCore::HTMLConstructionSite::reparent):
(WebCore::HTMLConstructionSite::insertAlreadyParsedChild):
(WebCore::HTMLConstructionSite::takeAllChildrenAndReparent):
(WebCore::HTMLConstructionSite::findFosterSite):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::HTMLTreeBuilder::processEndTag):
(WebCore::HTMLTreeBuilder::linkifyPhoneNumbers):
(WebCore::HTMLTreeBuilder::processTokenInForeignContent):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::logger const):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::TrackBase::setLogger):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::prepareToSpeak):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForEventListener):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::checkLoadComplete):
(WebCore::FrameLoader::loadItem):
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::HistoryController::goToItemShared):
(WebCore::HistoryController::recursiveSetProvisionalItem):
* Source/WebCore/loader/ImageLoader.cpp:
(WebCore::ImageLoader::updatedHasPendingEvent):
* Source/WebCore/loader/ProgressTracker.cpp:
(WebCore::ProgressTracker::progressStarted):
* Source/WebCore/loader/ResourceTimingInformation.cpp:
(WebCore::ResourceTimingInformation::addResourceTiming):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::tryReplaceEncodedData):
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::canRequestInContentDispositionAttachmentSandbox const):
* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::CachedResourceRequest::setInitiator):
(WebCore::CachedResourceRequest::updateForAccessControl):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenu):
(WebCore::prepareContextForQRCode):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::createTargetOriginForPostMessage):
* Source/WebCore/page/ElementTargetingController.cpp:
(WebCore::findChild):
(WebCore::targetedElementInfo):
(WebCore::findOnlyMainElement):
(WebCore::ElementTargetingController::extractTargets):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusNavigationScope::scopeOf):
(WebCore::FocusController::findFocusCandidateInContainer):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::FrameTree::appendChild):
(WebCore::FrameTree::replaceChild):
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::open):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::findMatchesForText):
(WebCore::Page::setDefersLoading):
(WebCore::Page::lockAllOverlayScrollbarsToHidden):
(WebCore::Page::setIsInWindowInternal):
(WebCore::Page::shouldUpdateAccessibilityRegions const):
(WebCore::Page::setDebugger):
(WebCore::Page::setMemoryCacheClientCallsEnabled):
(WebCore::Page::stopKeyboardScrollAnimation):
(WebCore::Page::suspendActiveDOMObjectsAndAnimations):
(WebCore::Page::resumeActiveDOMObjectsAndAnimations):
(WebCore::Page::setSessionID):
(WebCore::Page::idbConnection):
(WebCore::Page::forEachDocumentFromMainFrame):
(WebCore::Page::findMatchingLocalDocument const):
(WebCore::Page::forEachRenderableDocument const):
(WebCore::Page::forEachLocalFrame):
(WebCore::Page::forEachWindowEventLoop):
(WebCore::dispatchPrintEvent):
(WebCore::Page::startApplePayAMSUISession):
(WebCore::Page::forceRepaintAllFrames):
(WebCore::Page::reloadExecutionContextsForOrigin const):
(WebCore::Page::activeImmersiveXRSession const):
* Source/WebCore/page/PerformanceObserver.cpp:
(WebCore::PerformanceObserver::PerformanceObserver):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::dispatchEnterOrLeaveEvent):
(WebCore::PointerCaptureController::pointerEventWillBeDispatched):
* Source/WebCore/page/Quirks.cpp:
(WebCore::isKinjaLoginAvatarElement):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::ruleListFilter):
* Source/WebCore/page/cocoa/PageCocoa.mm:
(WebCore::Page::addSchedulePair):
(WebCore::Page::removeSchedulePair):
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::willNotProceedWithClick):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::frameViewForScrollingNode const):
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::findTextSuggestionMarkerByID const):
(WebCore::WritingToolsController::findTextSuggestionMarkerContainingRange const):
* Source/WebCore/platform/audio/AudioBus.cpp:
(WebCore::AudioBus::createBySampleRateConverting):

Canonical link: <a href="https://commits.webkit.org/296098@main">https://commits.webkit.org/296098@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/73762a20f8569017b4ab14aac119477b44ae2f80

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107340 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27022 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112554 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57876 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109304 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27704 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35522 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81483 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110269 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21950 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96743 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61857 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21389 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14874 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57319 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91319 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14907 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115655 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34406 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25357 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90525 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34782 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92993 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90261 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35169 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12952 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30150 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17364 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34328 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39865 "Failed to build and analyze WebKit") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34074 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37429 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35735 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -1271,7 +1271,7 @@ void Document::invalidateQuerySelectorAllResults(Node& startingNode)
     if (m_querySelectorAllResults.isEmptyIgnoringNullReferences())
         return;
-    for (RefPtr<Node> currentNode = &startingNode; currentNode; currentNode = currentNode->parentNode()) {
+    for (RefPtr<Node> currentNode = startingNode; currentNode; currentNode = currentNode->parentNode()) {
```
```suggestion
    for (RefPtr currentNode = startingNode; currentNode; currentNode = currentNode->parentNode()) {
```

annevk commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -1283,7 +1283,7 @@ void Document::invalidateQuerySelectorAllResultsForClassAttributeChange(Node& st
     if (m_querySelectorAllResults.isEmptyIgnoringNullReferences())
         return;
-    for (RefPtr<Node> currentNode = &startingNode; currentNode; currentNode = currentNode->parentNode()) {
+    for (RefPtr<Node> currentNode = startingNode; currentNode; currentNode = currentNode->parentNode()) {
```
```suggestion
    for (RefPtr currentNode = startingNode; currentNode; currentNode = currentNode->parentNode()) {
```

annevk commented on Source/WebCore/editing/ApplyStyleCommand.cpp:
```diff
@@ -775,7 +775,7 @@ void ApplyStyleCommand::applyInlineStyleToNodeRange(EditingStyle& style, Node& s
 
     Vector<InlineRunToApplyStyle> runs;
-    RefPtr<Node> node = &startNode;
+    RefPtr<Node> node = startNode;
```
```suggestion
    RefPtr node = startNode;
```

annevk commented on Source/WebCore/editing/ApplyStyleCommand.cpp:
```diff
@@ -667,7 +667,7 @@ void ApplyStyleCommand::applyInlineStyle(EditingStyle& style)
     document().updateLayoutIgnorePendingStylesheets();
 
-    RefPtr<EditingStyle> styleToApply = &style;
+    RefPtr<EditingStyle> styleToApply = style;
```
```suggestion
    RefPtr styleToApply = style;
```

annevk commented on Source/WebCore/editing/ApplyStyleCommand.cpp:
```diff
@@ -1347,7 +1347,7 @@ bool ApplyStyleCommand::surroundNodeRangeWithElement(Node& startNode, Node& endN
     }
 
-    RefPtr<Node> node = &startNode;
+    RefPtr<Node> node = startNode;
```
```suggestion
    RefPtr node = startNode;
```

annevk commented on Source/WebCore/editing/ApplyStyleCommand.cpp:
```diff
@@ -1427,8 +1427,8 @@ Position ApplyStyleCommand::positionToComputeInlineStyleChange(Node& startNode,
-    RefPtr<Node> startNode = &passedStart;
-    RefPtr<Node> endNode = &passedEnd;
+    RefPtr<Node> startNode = passedStart;
+    RefPtr<Node> endNode = passedEnd;
```
```suggestion
    RefPtr startNode = passedStart;
    RefPtr endNode = passedEnd;
```

cdumez committed:
Leverage the `RefPtr<T>(T&)` constructor in more places
https://bugs.webkit.org/show_bug.cgi?id=294309

Reviewed by Anne van Kesteren.

* Source/WebCore/Modules/WebGPU/Implementation/WebGPUCompositorIntegrationImpl.h:
* Source/WebCore/Modules/applepay/PaymentCoordinator.cpp:
(WebCore::PaymentCoordinator::beginPaymentSession):
* Source/WebCore/Modules/highlight/AppHighlightStorage.cpp:
(WebCore::findNodeStartingAtPathComponentIndex):
* Source/WebCore/Modules/indexeddb/IDBDatabase.cpp:
(WebCore::IDBDatabase::startVersionChangeTransaction):
* Source/WebCore/Modules/indexeddb/IDBRequest.cpp:
(WebCore::IDBRequest::setVersionChangeTransaction):
(WebCore::IDBRequest::dispatchEvent):
(WebCore::IDBRequest::willIterateCursor):
* Source/WebCore/Modules/indexeddb/IDBTransaction.cpp:
(WebCore::IDBTransaction::abortDueToFailedRequest):
(WebCore::IDBTransaction::completeNoncursorRequest):
(WebCore::IDBTransaction::completeCursorRequest):
* Source/WebCore/Modules/indexeddb/server/MemoryIndex.cpp:
(WebCore::IDBServer::MemoryIndex::writeTransactionStarted):
* Source/WebCore/Modules/indexeddb/server/MemoryObjectStore.cpp:
(WebCore::IDBServer::MemoryObjectStore::writeTransactionStarted):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::savePreviouslySelectedTextTrackIfNecessary):
* Source/WebCore/Modules/mediasession/MediaSession.cpp:
(WebCore::MediaSession::MediaSession):
* Source/WebCore/Modules/mediasession/MediaSessionCoordinatorPrivate.cpp:
(WebCore::MediaSessionCoordinatorPrivate::setLogger):
* Source/WebCore/Modules/mediastream/ImageCapture.cpp:
* Source/WebCore/Modules/mediastream/RTCEncodedFrame.cpp:
(WebCore::RTCEncodedFrame::setData):
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.cpp:
(WebCore::RTCRtpScriptTransform::initializeTransformer):
* Source/WebCore/Modules/speech/SpeechRecognition.cpp:
(WebCore::SpeechRecognition::SpeechRecognition):
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.h:
* Source/WebCore/Modules/webaudio/AudioWorkletNode.cpp:
(WebCore::AudioWorkletNode::setProcessor):
* Source/WebCore/Modules/webaudio/AudioWorkletNode.h:
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
(WebCore::OscillatorNode::setPeriodicWave):
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
(WebCore::WebCodecsVideoFrame::clone):
* Source/WebCore/accessibility/AXCoreObject.cpp:
(WebCore::AXCoreObject::partialOrder):
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::AXObjectCache::handleChildrenChanged):
* Source/WebCore/accessibility/AXTextMarker.cpp:
(WebCore::Accessibility::findObjectWithRuns):
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper accessibilityRowRange]):
* Source/WebCore/accessibility/playstation/AXObjectCachePlayStation.cpp:
(WebCore::AXObjectCache::postPlatformNotification):
* Source/WebCore/animation/KeyframeEffect.cpp:
(WebCore::KeyframeEffect::setTarget):
* Source/WebCore/animation/StyleOriginatedTimelinesController.cpp:
(WebCore::StyleOriginatedTimelinesController::determineTreeOrder):
(WebCore::StyleOriginatedTimelinesController::updateTimelineForTimelineScope):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::collectIsolatedContexts):
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
(WebCore::ScriptModuleLoader::fetch):
* Source/WebCore/bindings/js/WorkerScriptFetcher.h:
* Source/WebCore/css/CSSFontFaceSrcValue.cpp:
(WebCore::CSSFontFaceSrcLocalValue::setSVGFontFaceElement):
* Source/WebCore/css/ShorthandSerializer.cpp:
(WebCore::ShorthandSerializer::serializeBorderRadius const):
* Source/WebCore/css/calc/CSSCalcTree+ContainerProgressEvaluator.cpp:
(WebCore::CSSCalc::evaluateContainerProgress):
* Source/WebCore/dom/Attr.cpp:
(WebCore::Attr::attachToElement):
* Source/WebCore/dom/ChildListMutationScope.cpp:
(WebCore::ChildListMutationAccumulator::childAdded):
* Source/WebCore/dom/ContainerNode.cpp:
(WebCore::containsIncludingHostElements):
(WebCore::dispatchChildInsertionEvents):
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
(WebCore::notifyNodeInsertedIntoDocument):
(WebCore::notifyNodeInsertedIntoTree):
(WebCore::notifyNodeRemovedFromDocument):
(WebCore::notifyNodeRemovedFromTree):
* Source/WebCore/dom/ContentVisibilityDocumentState.cpp:
(WebCore::ContentVisibilityDocumentState::updateContentRelevancyForScrollIfNeeded):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::invalidateQuerySelectorAllResults):
(WebCore::Document::invalidateQuerySelectorAllResultsForClassAttributeChange):
(WebCore::Document::processBaseElement):
(WebCore::Document::idbConnectionProxy):
* Source/WebCore/dom/DocumentFullscreen.cpp:
(WebCore::DocumentFullscreen::willEnterFullscreen):
(WebCore::documentsToUnfullscreen):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::topmostPopoverAncestor):
* Source/WebCore/dom/Event.cpp:
(WebCore::Event::setEventPath):
(WebCore::Event::timeStampForBindings const):
* Source/WebCore/dom/EventPath.cpp:
(WebCore::RelatedNodeRetargeter::RelatedNodeRetargeter):
* Source/WebCore/dom/FragmentDirectiveRangeFinder.cpp:
(WebCore::FragmentDirectiveRangeFinder::rangeOfStringInRange):
* Source/WebCore/dom/FragmentDirectiveUtilities.cpp:
(WebCore::FragmentDirectiveUtilities::nearestBlockAncestor):
* Source/WebCore/dom/InternalObserverDrop.cpp:
* Source/WebCore/dom/InternalObserverFilter.cpp:
* Source/WebCore/dom/InternalObserverMap.cpp:
* Source/WebCore/dom/InternalObserverTake.cpp:
* Source/WebCore/dom/LoadableModuleScript.cpp:
(WebCore::LoadableModuleScript::notifyLoadCompleted):
* Source/WebCore/dom/Range.cpp:
(WebCore::Range::surroundContents):
* Source/WebCore/dom/RejectedPromiseTracker.cpp:
(WebCore::RejectedPromiseTracker::reportUnhandledRejections):
* Source/WebCore/dom/UserGestureIndicator.cpp:
(WebCore::UserGestureToken::UserGestureToken):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyInlineStyle):
(WebCore::ApplyStyleCommand::applyInlineStyleToNodeRange):
(WebCore::ApplyStyleCommand::surroundNodeRangeWithElement):
(WebCore::ApplyStyleCommand::applyInlineStyleChange):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::wrappingStyleForSerialization):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::appliedEditing):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::DragCaretController::setCaretPosition):
(WebCore::FrameSelection::associateLiveRange):
* Source/WebCore/editing/MarkupAccumulator.cpp:
(WebCore::MarkupAccumulator::serializeNodesWithNamespaces):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::nextInPreOrderCrossingShadowBoundaries):
(WebCore::shouldEmitNewlineAfterNode):
(WebCore::TextIterator::emitText):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::isInUserAgentShadowRootOrHasEditableShadowAncestor):
* Source/WebCore/editing/markup.cpp:
(WebCore::UserSelectNoneStateCache::computeState):
(WebCore::StyledMarkupAccumulator::traverseNodesForSerialization):
(WebCore::highestAncestorToWrapMarkup):
* Source/WebCore/fileapi/FileReader.cpp:
(WebCore::FileReader::readInternal):
* Source/WebCore/html/HTMLFrameOwnerElement.cpp:
(WebCore::SubframeLoadingDisabler::canLoadFrame):
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::checkedRadioButtonForGroup const):
* Source/WebCore/html/HTMLMapElement.cpp:
(WebCore::HTMLMapElement::mapMouseEvent):
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::speakCueText):
(WebCore::HTMLMediaElement::sourceWasAdded):
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::recalcListItems const):
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::setContainerAndOffsetForRange):
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::computeIsDisabledByFieldsetAncestor const):
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::beginQuery):
* Source/WebCore/html/canvas/WebGLTransformFeedback.cpp:
(WebCore::WebGLTransformFeedback::setProgram):
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
(WebCore::insert):
(WebCore::HTMLConstructionSite::insertTextNode):
(WebCore::HTMLConstructionSite::reparent):
(WebCore::HTMLConstructionSite::insertAlreadyParsedChild):
(WebCore::HTMLConstructionSite::takeAllChildrenAndReparent):
(WebCore::HTMLConstructionSite::findFosterSite):
* Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
(WebCore::HTMLTreeBuilder::processEndTag):
(WebCore::HTMLTreeBuilder::linkifyPhoneNumbers):
(WebCore::HTMLTreeBuilder::processTokenInForeignContent):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::logger const):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::TrackBase::setLogger):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::prepareToSpeak):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::buildObjectForEventListener):
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::checkLoadComplete):
(WebCore::FrameLoader::loadItem):
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::HistoryController::goToItemShared):
(WebCore::HistoryController::recursiveSetProvisionalItem):
* Source/WebCore/loader/ImageLoader.cpp:
(WebCore::ImageLoader::updatedHasPendingEvent):
* Source/WebCore/loader/ProgressTracker.cpp:
(WebCore::ProgressTracker::progressStarted):
* Source/WebCore/loader/ResourceTimingInformation.cpp:
(WebCore::ResourceTimingInformation::addResourceTiming):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::tryReplaceEncodedData):
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::canRequestInContentDispositionAttachmentSandbox const):
* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::CachedResourceRequest::setInitiator):
(WebCore::CachedResourceRequest::updateForAccessControl):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenu):
(WebCore::prepareContextForQRCode):
* Source/WebCore/page/DOMWindow.cpp:
(WebCore::DOMWindow::createTargetOriginForPostMessage):
* Source/WebCore/page/ElementTargetingController.cpp:
(WebCore::findChild):
(WebCore::targetedElementInfo):
(WebCore::findOnlyMainElement):
(WebCore::ElementTargetingController::extractTargets):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusNavigationScope::scopeOf):
(WebCore::FocusController::findFocusCandidateInContainer):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::FrameTree::appendChild):
(WebCore::FrameTree::replaceChild):
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::open):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::findMatchesForText):
(WebCore::Page::setDefersLoading):
(WebCore::Page::lockAllOverlayScrollbarsToHidden):
(WebCore::Page::setIsInWindowInternal):
(WebCore::Page::shouldUpdateAccessibilityRegions const):
(WebCore::Page::setDebugger):
(WebCore::Page::setMemoryCacheClientCallsEnabled):
(WebCore::Page::stopKeyboardScrollAnimation):
(WebCore::Page::suspendActiveDOMObjectsAndAnimations):
(WebCore::Page::resumeActiveDOMObjectsAndAnimations):
(WebCore::Page::setSessionID):
(WebCore::Page::idbConnection):
(WebCore::Page::forEachDocumentFromMainFrame):
(WebCore::Page::findMatchingLocalDocument const):
(WebCore::Page::forEachRenderableDocument const):
(WebCore::Page::forEachLocalFrame):
(WebCore::Page::forEachWindowEventLoop):
(WebCore::dispatchPrintEvent):
(WebCore::Page::startApplePayAMSUISession):
(WebCore::Page::forceRepaintAllFrames):
(WebCore::Page::reloadExecutionContextsForOrigin const):
(WebCore::Page::activeImmersiveXRSession const):
* Source/WebCore/page/PerformanceObserver.cpp:
(WebCore::PerformanceObserver::PerformanceObserver):
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::PointerCaptureController::dispatchEnterOrLeaveEvent):
(WebCore::PointerCaptureController::pointerEventWillBeDispatched):
* Source/WebCore/page/Quirks.cpp:
(WebCore::isKinjaLoginAvatarElement):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::ruleListFilter):
* Source/WebCore/page/cocoa/PageCocoa.mm:
(WebCore::Page::addSchedulePair):
(WebCore::Page::removeSchedulePair):
* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::willNotProceedWithClick):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::frameViewForScrollingNode const):
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::findTextSuggestionMarkerByID const):
(WebCore::WritingToolsController::findTextSuggestionMarkerContainingRange const):
* Source/WebCore/platform/audio/AudioBus.cpp:
(WebCore::AudioBus::createBySampleRateConverting):

Canonical link: https://commits.webkit.org/296098@main
